### PR TITLE
IPython improvements

### LINKED ISSUE #520: Make the `%%memray_flamegraph` IPython magic use aggregated capture files
See #519 for context. The `%%memray_flamegraph` IPython magic predates [aggregated capture files](https://bloomberg.github.io/memray/run.html#aggregated-capture-files), but considering that the capture file produced by `%%memray_flamegraph` is only used for the reports that aggregated capture files support, we should probably use aggregated capture files by default for the `%%memray_flamegraph` magic.

### PR DESCRIPTION
**Pull request by godlygeek**
Closes #520 
See #519

### CODE REVIEW

File: src/memray/_ipython/flamegraph.py [hunk 1]
```diff
@@ -118,6 +133,15 @@ def memray_flamegraph(self, line: str, cell: str) -> None:
...
             return
 
+        if options.temporal and options.temporary_allocation_threshold >= 0:
+            raise UsageError(
+                "Can't create a temporal flame graph of temporary allocations"
+            )
+        elif options.temporal or options.temporary_allocation_threshold >= 0:
+            file_format = "ALL_ALLOCATIONS"
+        else:
+            file_format = "AGGREGATED_ALLOCATIONS"
```
Comment (pablogsal) [inline src/memray/_ipython/flamegraph.py:143]: Ah, this will be a nice improvement 👍  (I think we were not doing it already)

**Review by pablogsal — APPROVED** [review]

File: tests/integration/test_ipython.py [hunk 2]
```diff
@@ -137,3 +137,35 @@ def test_passing_valid_argument(self, tmpdir, capsys):
...
+    @pytest.mark.parametrize(
+        "args,title",
+        [
+            ("--temporary-allocation-threshold=2", "flamegraph report"),
+            ("--leaks", "flamegraph report (memory leaks)"),
+            ("--temporal", "temporal flamegraph report"),
+            ("--leaks --temporal", "temporal flamegraph report (memory leaks)"),
+        ],
+    )
+    def test_report_title_by_report_type(self, tmpdir, capsys, args, title):
```
Comment (pablogsal) [inline tests/integration/test_ipython.py:150]: Maybe is worth checking the error when --temporal and `temporary_allocation_threshold>=0`

Comment (godlygeek) [reply tests/integration/test_ipython.py:150]: Ah, good call, can do.

File: src/memray/_ipython/flamegraph.py (hunk 1 above)
Comment (godlygeek) [reply src/memray/_ipython/flamegraph.py:143]: 💯

Comment (pablogsal) [reply src/memray/_ipython/flamegraph.py:143]: Now that I think about it we could do it in the pytest plugin as well 🤔

Comment (godlygeek) [reply src/memray/_ipython/flamegraph.py:143]: > Now that I think about it we could do it in the pytest plugin as well 🤔

https://github.com/bloomberg/pytest-memray/pull/107

**Pull request merged**